### PR TITLE
Make it possible to override permissions in django settings

### PR DESCRIPTION
**Pull request by eodolphi**

### CODE REVIEW

**Comment by sergei-maertens** [comment]
Can you make sure all tests pass?

Thanks for the README update by the way, I've been neglecting that too much.

File: exportdb/permissions.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import rules
-# TODO: this should be parametrized via settings
-
+from django.conf import settings
 
 @rules.predicate
-def is_superuser(user):
-    return user.is_superuser
+def permission(*args, **kwargs):
+    predicate = getattr(settings, 'EXPORTDB_PERMISSION', rules.is_superuser)
```
Comment (sergei-maertens) [inline exportdb/permissions.py]: I think we should rather do something like:

```
# in exportdb/conf.py
class ExportDBConf(AppConf):
    EXPORT_ROOT = os.path.join(settings.MEDIA_ROOT, EXPORT_SUBDIR)
    EXPORT_MEDIA_URL = posixpath.join(settings.MEDIA_URL, EXPORT_SUBDIR)
    EXPORT_CONF = {}
    PERMISSION = 'rules.is_superuser'  # add the default

    # form used in admin view to confirm export
    CONFIRM_FORM = 'django.forms.Form'
```

with

```
# in exportdb/permissions.py
...
from django.conf import settings

import rules

from .compat import import_string


@rules.predicate
def permission(*args, **kwargs):
    predicate = import_string(settings.EXPORTDB_PERMISSION)
    return predicate(*args, **kwargs)


rules.add_rule('exportdb.can_export', permission)
```

This should cover the desired flexibility for now and I'll have look at cleaner permission handling later. Rules is a very powerful library where you can logically combine predicates, we should definitely leverage that.

On a sidenote, please run pylint and PEP8 on the code :)

File: tests/app/tests/test_permissions.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        self.client.login(username='test', password='letmein')
+
+    def test_default_permission(self):
+        self.assertTrue(
+            rules.test_rule('exportdb.can_export', self.super_user)
+        )
+
+        self.assertFalse(
+            rules.test_rule('exportdb.can_export', self.normal_user)
+        )
```
Comment (sergei-maertens) [inline tests/app/tests/test_permissions.py:33]: Maybe add a test here with a staff user too, just to be sure

File: exportdb/permissions.py (hunk 1 above)
Comment (eodolphi) [inline exportdb/permissions.py]: I started out with something like this (without the appconf stuff, totally missed that). But we need `rules.is_group_member('something')` as a rule. Which will not work with `import_string`, that's why I choose this option.

Comment (sergei-maertens) [inline exportdb/permissions.py]: I'm assuming you're import `rules` in the settings file then?

**Comment by eodolphi** [comment]
Tests pass now. I was using an older version of mock that was in my virtualenv by accident. So I updated that dependency to.

I added a default to the appconf, but did not change the setting to a string. See comment above. If that is an absolute requirement, I can change it, and work around it on our side.

**Comment by sergei-maertens** [comment]
Let's just go with this version for now, we'll see if we can come up with a clever permissions systems later. I'll upload a minor update to PyPi in a minute.

**Pull request merged**